test/remittance.js

```diff
@@ -1,10 +1,438 @@
+        const cost = await getGasCost(txObj);
+        const expectedBrokerEthBalance = initBrokerEthBalance.add(toBN(2)).sub(cost).toString(10);
+        assert.equal(brokerEthBalance.toString(10), expectedBrokerEthBalance);
+
+    });
+
+    it("Funder reclaims funds from an expired remittance", async () => {
+
+        const expiryDate = futureTimeStamp();
+
+        //Create a remittance with an expired date
+        await remittance.create(hash, broker, expiryDate, {from: funder, value: 2});
+        const initFunderEthBalance = toBN(await getBalance(funder));
+        await timeMachine.advanceTimeAndBlock(day24hourinSecs);
+
+        const txObj = await remittance.reclaim(hash, {from: funder});
+
+        await truffleAssert.eventEmitted(txObj, 'RemittanceFundsReclaimed', (ev) => {
+            return  ev.hash === hash &&
+                    ev.funder === funder &&
+                    ev.amount.toString(10) === "2";
+        }, 'RemittanceFundsReclaimed event is emitted');
+
+        const remittanceInstance = await remittance.remittances(hash);
+        assert.strictEqual(remittanceInstance.fundsOwed.toString(10), "0");
+        assert.strictEqual(remittanceInstance.expiryDate.toString(10), "0");
+
+        // Check the remittance amount has been taken from the contract eth balance
+        const contractEthBalance = toBN(await getBalance(remittance.address));
+        assert.strictEqual(contractEthBalance.toString(10), "0");
+
+        // Check the remittance amount has been sent to the funder eth balance
+        const funderEthBalance = toBN(await getBalance(funder));
+        const cost = await getGasCost(txObj);
+        const expectedFunderEthBalance = initFunderEthBalance.add(toBN(2)).sub(cost).toString(10);
+        assert.strictEqual(funderEthBalance.toString(10), expectedFunderEthBalance);
+
+    });
+
+    it("Password cannot be reused to release in a second contract using the same creation hash", async () => {
+
+        // Creating by reusing the original contract's hash seeded with the original password and first contract's address.
+        // Releasing with the original password is unsuccessful
+        let expiryDate = futureTimeStamp();
+        const remittance2 = await Remittance.new({from: funder})
+        await remittance2.create(hash, broker, expiryDate, {from: funder, value: 2});
+
+        await truffleAssert.reverts(
+            remittance2.release(passwordBytes32, {from: broker}),
+            noFundsAvailableMsg
+        );
+
+        const result = await truffleAssert.createTransactionResult(remittance2, remittance2.transactionHash);
+        await truffleAssert.eventNotEmitted(
+            result
+        );
+
+        // Creating by using a new hash seeded with original password and the new contract's address.
+        // Releasing with the original password is successful.
+        // However this cannot be deduced from the transactions of the first contract.
+        expiryDate = futureTimeStamp();
```

Comment:
Better use another variable name, and make them both `const`. We are not looking for memory optimisations here.